workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/utilities/TokenBuilder.kt

```diff
@@ -0,0 +1,157 @@
+package com.r3.corda.lib.tokens.workflows.utilities
+
+import com.r3.corda.lib.tokens.contracts.states.FungibleToken
+import com.r3.corda.lib.tokens.contracts.types.IssuedTokenType
+import com.r3.corda.lib.tokens.contracts.types.TokenType
+import com.r3.corda.lib.tokens.contracts.utilities.amount
+import com.r3.corda.lib.tokens.contracts.utilities.heldBy
+import com.r3.corda.lib.tokens.contracts.utilities.issuedBy
+import com.r3.corda.lib.tokens.contracts.utilities.of
+import com.r3.corda.lib.tokens.workflows.TokenBuilderException
+import net.corda.core.contracts.Amount
+import net.corda.core.identity.Party
+import java.lang.Exception
+import java.math.BigDecimal
+
+/**
+ * A utility class designed for Java developers to more easily access Kotlin DSL
+ * functions to build token types.
+ *
+ * This function vaguely follows a builder pattern, design choices were made
+ * to emulate Kotlin syntax as closely as possible for an easily transferable
+ * developer experience in Java.
+ */
+class TokenBuilder {
+    private var longAmount: Long? = null
```

Comment:
Maybe lets use BigDecimal as the lowest common dinaminator for all other data types? Then there won't be a need for Lond, double, etc.

Reply:
Very interesting, I'll try this one out.